test/swe2d/test_standing_wave.py

```diff
@@ -0,0 +1,104 @@
+# Test for temporal convergence of cranknicolson and pressureprojection picard timesteppers,
+# tests convergence of a single period of a standing wave in a rectangular channel.
+# This only tests against a linear solution, so does not really test whether the splitting
+# in pressureprojectionpicard between nonlinear momentum and linearized wave equation terms is correct.
+# pressureprojectionpicard does need two iterations to ensure 2nd order convergence
+from thetis import *
+import pytest
+import math
+
+
+@pytest.mark.parametrize("timesteps,max_rel_err", [
+    (10, 0.02), (20, 5e-3)])
+# with nonlin=True and nx=100 this converges for the series
+#  (10,0.02), (20,5e-3), (40, 1.25e-3)
+# with nonlin=False further converge is possible
+@pytest.mark.parametrize("timestepper", [
+    'cranknicolson', 'pressureprojectionpicard', ])
+def test_steady_state_channel(timesteps, max_rel_err, timestepper, do_export=False):
+
+    lx = 5e3
+    ly = 1e3
+    nx = 50
+    mesh2d = RectangleMesh(nx, 1, lx, ly)
+
+    n = timesteps
+    depth = 100.
+    g = physical_constants['g_grav'].dat.data[0]
+    c = math.sqrt(g*depth)
+    period = 2*lx/c
+    dt = period/n
+    t_end = period-0.1*dt  # make sure we don't overshoot
+
+    x = SpatialCoordinate(mesh2d)
+    elev_init = cos(pi*x[0]/lx)
+
+    # bathymetry
+    p1_2d = FunctionSpace(mesh2d, 'CG', 1)
+    bathymetry_2d = Function(p1_2d, name="bathymetry")
+    bathymetry_2d.assign(depth)
+
+    # --- create solver ---
+    solver_obj = solver2d.FlowSolver2d(mesh2d, bathymetry_2d, order=1)
+    solver_obj.options.nonlin = True
```

Comment:
Yes, indeed. However with a small amplitude (wrt depth) within a single period it should remain very very close to the linear solution (see also comments on top). Not really testing the nonlinear terms therefore (just that the split doesn't mess things up).